vars/neoDeploy.groovy

```diff
@@ -1,94 +1,85 @@
-        'deployAccount', //deprecated, replaced by parameter 'account'
-        'deployHost', //deprecated, replaced by parameter 'host'
-        'deployMode',
-        'dockerEnvVars',
-        'dockerImage',
-        'dockerOptions',
-        'host',
-        'neoCredentialsId',
-        'neoHome',
-        'propertiesFile',
-        'runtime',
-        'runtimeVersion',
-        'vmSize',
-        'warAction'
-        ]
-
-    Set stepConfigurationKeys = [
-        'account',
-        'dockerEnvVars',
-        'dockerImage',
-        'dockerOptions',
-        'host',
-        'neoCredentialsId',
-        'neoHome'
-        ]
-
-    handlePipelineStepErrors (stepName: stepName, stepParameters: parameters) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
 
         def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
-
         def utils = new Utils()
 
         prepareDefaultValues script: script
 
-        final Map stepConfiguration = [:]
+        final Map stepCompatibilityConfiguration = [:]
 
         // Backward compatibility: ensure old configuration is taken into account
         // The old configuration in not stage / step specific
 
         def defaultDeployHost = script.commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
         if(defaultDeployHost) {
-            echo "[WARNING][${stepName}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_HOST'. This configuration framework will be removed in future versions."
-            stepConfiguration.put('host', defaultDeployHost)
+            echo "[WARNING][${STEP_NAME}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_HOST'. This configuration framework will be removed in future versions."
+            stepCompatibilityConfiguration.put('host', defaultDeployHost)
         }
 
         def defaultDeployAccount = script.commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
         if(defaultDeployAccount) {
-            echo "[WARNING][${stepName}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_ACCOUNT'. This configuration framekwork will be removed in future versions."
-            stepConfiguration.put('account', defaultDeployAccount)
+            echo "[WARNING][${STEP_NAME}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_ACCOUNT'. This configuration framekwork will be removed in future versions."
+            stepCompatibilityConfiguration.put('account', defaultDeployAccount)
         }
 
         if(parameters.deployHost && !parameters.host) {
-            echo "[WARNING][${stepName}] Deprecated parameter 'deployHost' is used. This will not work anymore in future versions. Use parameter 'host' instead."
+            echo "[WARNING][${STEP_NAME}] Deprecated parameter 'deployHost' is used. This will not work anymore in future versions. Use parameter 'host' instead."
```

Comment:
Why do we emit a warning here only and do not use the ```stepCompatibilityConfiguration``` like above?

Reply:
the steps above refer to the config file, this refers to the `parameters`

Reply:
But why can't we say ```stepCompatibilityConfiguration.put('host', parameters.deployHost)``` ?

With this the new property ```host``` would be mixed-in with the value from ```deployHost```. Afterwards we have - of course - to use ```host```. What would be wrong with this approach?